flink-java/src/main/java/org/apache/flink/api/java/ExecutionEnvironment.java

```diff
@@ -911,6 +912,24 @@ public JobExecutionResult execute() throws Exception {
 	public abstract JobExecutionResult execute(String jobName) throws Exception;
 
 	/**
+	 * Triggers the program execution, just like {@code execute()} but does not block.
+	 * Instead, it returns a JobClient which can be used to interact with the running job.
+	 * @return A JobClient for job interaction.
+	 * @throws Exception Thrown if the program submission fails.
+	 */
+	public JobClient executeWithControl() throws Exception {
+		return executeWithControl(getDefaultName());
+	}
+
+	/**
+	 * Triggers the program execution, just like {@code execute(String jobName)} but does not block.
+	 * Instead, it returns a JobClient which can be used to interact with the running job.
+	 * @return A JobClient for job interaction.
+	 * @throws Exception Thrown if the program submission fails.
+	 */
+	public abstract JobClient executeWithControl(String jobName) throws Exception;
```

Comment:
Names for new concepts are always hard. I don't think `executeAttached` is any better than `executeWithControl`. `submitJob` would introduce the job concept into the API. I think the new name should definitely start with `execute` to follow the existing convention and be easy discoverable.